core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1315 @@
+    /// Includes the the fee paid for the registration, and the expiration time
+    struct RegisterNameEvent has drop, store {
+        domain_name: String,
+        subdomain_name: Option<String>,
+        registration_fee_octas: u64,
+        expiration_time_secs: u64,
+    }
+
+    /// A name (potentially subdomain) has been renewed on chain
+    /// Includes the the fee paid for the registration, and the expiration time
+    struct RenewNameEvent has drop, store {
+        domain_name: String,
+        subdomain_name: Option<String>,
+        renewal_fee_octas: u64,
+        expiration_time_secs: u64,
+
+        // Extras for indexing
+        target_address: Option<address>,
+        is_primary_name: bool,
+    }
+
+    /// This is only callable during publishing
+    fun init_module(account: &signer) {
+        let funds_address: address = @aptos_names_funds;
+        let admin_address: address = @aptos_names_admin;
+
+        if (!account::exists_at(funds_address)) {
+            aptos_account::create_account(funds_address);
+        };
+
+        if (!account::exists_at(admin_address)) {
+            aptos_account::create_account(admin_address);
+        };
+
+        v2_config::initialize_config(account, admin_address, funds_address);
+
+        // Create collection + token_resource
+        let constructor_ref = object::create_named_object(
+            account,
+            APP_OBJECT_SEED,
+        );
+        let extend_ref = object::generate_extend_ref(&constructor_ref);
+        let app_signer = &object::generate_signer(&constructor_ref);
+        collection::create_unlimited_collection(
+            app_signer,
+            utf8(COLLECTION_DESCRIPTION),
+            v2_config::domain_collection_name(),
+            option::none(),
+            utf8(COLLECTION_URI),
+        );
+        collection::create_unlimited_collection(
+            app_signer,
+            utf8(SUBDOMAIN_COLLECTION_DESCRIPTION),
+            v2_config::subdomain_collection_name(),
+            option::none(),
+            utf8(COLLECTION_URI),
+        );
+        aptos_account::create_account(signer::address_of(app_signer));
+        move_to(app_signer, SetTargetAddressEvents {
+            set_name_events: account::new_event_handle<SetTargetAddressEvent>(app_signer),
+        });
+        move_to(app_signer, RegisterNameEvents {
+            register_name_events: account::new_event_handle<RegisterNameEvent>(app_signer),
+        });
+        move_to(app_signer, RenewNameEvents {
+            renew_name_events: account::new_event_handle<RenewNameEvent>(app_signer),
+        });
+        move_to(app_signer, SetReverseLookupEvents {
+            set_reverse_lookup_events: account::new_event_handle<SetReverseLookupEvent>(app_signer),
+        });
+        move_to(app_signer, DomainObject {
+            extend_ref,
+        });
+    }
+
+    /// Creates a token for the name.
+    /// NOTE: This function performs no validation checks
+    fun create_token(
```

Comment:
feels like create_token should not have any logic for subdomain and we should have a function that wraps this for create_domain and create_subdomain with whatever additional logic for example for subdomains we should have ... let's see what happens in the rest of the code.